### PR TITLE
Add character-based LOO

### PR DESCRIPTION
**Pull request by eveenhuis**
Adds character based LOO. Removes duplicate ICL examples, based on configured similarity measure. Phase 1 experiment directory created.

Note:
- `incontext.leave_one_out=false`  should now be specified as `incontext.leave_one_out=null`
- `incontext.leave_one_out=true`  should now be specified as `incontext.leave_one_out=scenario_description` (old behavior) or `incontext.leave_one_out=characters`

### CODE REVIEW

**Review by dmjoy — APPROVED** [review]
Seems fine to me, minor preference for making the parameter `incontext.leave_one_out_strategy` as the current name strongly implies a boolean (that and changing the parameter name will hopefully break more loudly and obviously if someone tries `incontext.leave_one_out=true` etc. moving forward since that parameter won't exist)

File: align_system/algorithms/oracle_adm.py
Comment (jadie1) [inline align_system/algorithms/oracle_adm.py:1]: @dmjoy This is the same fix I've been using to handle the targets in dictionary format

File: align_system/configs/experiment/phase1_evaluation/random_adept_ingroup_bias_train.yaml
Comment (jadie1) [inline align_system/configs/experiment/phase1_evaluation/random_adept_ingroup_bias_train.yaml:1]: We may want to add `force_determinism` for the random runs

**Review by jadie1 — APPROVED** [review]
Looks good to me!

File: align_system/configs/experiment/phase1_evaluation/random_adept_ingroup_bias_train.yaml
Comment (eveenhuis) [reply align_system/configs/experiment/phase1_evaluation/random_adept_ingroup_bias_train.yaml:1]: Then they aren't random :) I tend to use the random experiments as a base for my other experiments (so I wouldn't prefer the change), but I can make the change if we think it makes sense

**Comment by eveenhuis** [comment]
> Seems fine to me, minor preference for making the parameter `incontext.leave_one_out_strategy` as the current name strongly implies a boolean (that and changing the parameter name will hopefully break more loudly and obviously if someone tries `incontext.leave_one_out=true` etc. moving forward since that parameter won't exist)

@dmjoy done!

**Comment by dmjoy** [comment]
Thanks @eveenhuis, merging in just a few

File: align_system/configs/experiment/phase1_evaluation/random_adept_ingroup_bias_train.yaml
Comment (jadie1) [reply align_system/configs/experiment/phase1_evaluation/random_adept_ingroup_bias_train.yaml:1]: I'm fine to leave it then. I guess if we wanted to truly benchmark random performance we'd have to average across multiple runs without determinism

**Pull request merged**